docs/sdk/architecture.md

```diff
@@ -0,0 +1,84 @@
+---
+title: System Architecture
+sidebar_position: 2
+---
+
+## Overview
+
+When building DevCycle, we decided on a couple of core design goals from our experience of building global services and SDKs:
+- SDKs should be easy to understand and have consistent functionality across platforms.
+- Feature Flags should evaluate quickly and served from as close to the end user as possible.
+- Business-critical code should be shared across platforms to reduce the amount we need to maintain.
+- A cross-platform end-to-end test-harness is required to ensure SDKs are working as expected. 
+- SDKs should limit the number of start options that change the core behaviour of the SDK.
+
+With these design goals in mind, we have developed a set of Client / Mobile / Server SDKs that deliver 
+fast global response times and local evaluation of feature flag values.
+
```

Comment:
Thinking more - i'd add a section here saying what role WASM has and where we use it and where we don't